allennlp/data/dataset_readers/wikitables/wikitables_dataset_reader.py

```diff
@@ -0,0 +1,171 @@
+    the expected format). If you are doing the former, you still need to provide a ``tables_directory`` path here.
+
+    For training, we assume you are reading in ``data/*.examples`` files, and you have access to the output from
+    Dynamic Programming on Denotations (DPD) on the training dataset.
+
+    Parameters
+    ----------
+    tables_directory : ``str`` (optional)
+        Prefix for the path to the directory in which the tables reside. For example, ``*.examples`` files
+        contain paths like ``csv/204-csv/590.csv``, this is the directory that contains the ``csv`` directory.
+    dpd_output_directory : ``str`` (optional)
+        Directory that contains all the gzipped dpd output files. We assume the filenames match the
+        example IDs (e.g.: ``nt-0.gz``).
+    utterance_tokenizer : ``Tokenizer`` (optional)
+        Tokenizer to use for the questions. Will default to ``WordTokenizer()``.
+    utterance_token_indexers : ``Dict[str, TokenIndexer]`` (optional)
+        Token indexers for questions. Will default to ``{"tokens": SingleIdTokenIndexer()}``.
+    """
+    def __init__(self,
+                 tables_directory: str = None,
+                 dpd_output_directory: str = None,
+                 utterance_tokenizer: Tokenizer = None,
+                 utterance_token_indexers: Dict[str, TokenIndexer] = None) -> None:
+        self._tables_directory = tables_directory
+        self._dpd_output_directory = dpd_output_directory
+        self._utterance_tokenizer = utterance_tokenizer or WordTokenizer()
+        self._utterance_token_indexers = utterance_token_indexers or {"tokens": SingleIdTokenIndexer()}
+
+    @overrides
+    def read(self, file_path):
+        instances = []
+        with open(file_path, "r") as data_file:
+            logger.info("Reading instances from lines in file: %s", file_path)
+            for line in tqdm.tqdm(data_file):
+                line = line.strip("\n")
+                if not line:
+                    continue
+                parsed_info = self._parse_line_as_lisp(line)
+                utterance = parsed_info["utterance"]
+                # We want the TSV file, but the ``*.examples`` files typically point to CSV.
+                table_filename = "%s/%s" % (self._tables_directory, parsed_info["context"].replace(".csv", ".tsv"))
+                dpd_output_filename = "%s/%s.gz" % (self._dpd_output_directory, parsed_info["id"])
+                sempre_forms = [line.strip().decode('utf-8') for line in gzip.open(dpd_output_filename)]
+                instances.append(self.text_to_instance(utterance, table_filename, sempre_forms))
+        if not instances:
+            raise ConfigurationError("No instances read!")
+        return Dataset(instances)
+
+    @overrides
+    def text_to_instance(self,  # type: ignore
+                         utterance: str,
+                         table_info: TableInfoType,
```

Comment:
You can just do `Union[str, JsonDict]` here, instead of creating a type that's only used once.  I'd recommend the same for your `LispValueType`.

Reply:
Done.